conjecture-rust/src/engine.rs

```diff
@@ -59,13 +61,22 @@ impl MainGenerationLoop {
     }
 
     fn loop_body(&mut self) -> StepResult {
-        let interesting_example = self.generate_examples()?;
-
-        let mut shrinker = Shrinker::new(self, interesting_example, |r| {
-            r.status == Status::Interesting
-        });
-
-        shrinker.run()?;
+        self.generate_examples()?;
+
+        while self.minimized_examples.len() > self.fully_minimized.len() {
```

Comment:
It's unclear to me why we'd iterate more than once through this loop. Shouldn't inserting all keys from `minimized_examples` into `fully_minimized` ensure that they end up with the same size after the first iteration?